lib/matplotlib/category.py

```diff
@@ -183,6 +186,21 @@ def update(self, data):
         """
         data = np.atleast_1d(np.array(data, dtype=object))
 
+        convertable = True
+        for val in data:
```

Comment:
I think this should be in the ` for val in OrderedDict.fromkeys(data):` loop before. It would fix some of the problems of the tests raising warnings and also errors because then the typeerror would trigger before the conversion warning.

Reply:
@story645.  Why is data put in an OrderedDict there?   The check could go there, but I’d didn’t understand what that did so avoided that loop.

Reply:
it's the recommend way to get an unsorted list of the unique values in a iterable.

Reply:
huh, OK, why do they need to be unique?  I can't imagine the overhead of making a dict and then unpacking it is worth it compared to a few repeated checks, not to mention that its pretty inscrutable, at least to me...

Reply:
Because the next couple of lines build the string->float mapping and it's useful for that task to not have to double check that the val is not in the mapping. You're welcome to add a comment explaining the ordered dict stuff

Reply:
I still don't see how this helps.  The following lines check if `val` is in the dict, and if you've already added it nothing happens.  Taking this out, and just iterating over `data` doesn't fail any tests, but if there is a case that will fail, let me know and I'll add to tests and put the code back the way it was.

Reply:
Please put the orderedict back. It's there cause for larger inputs, it's more efficient to not have to go through the whole set. Do the tests still pass if you keep orderedict and take out the membership check?

Reply:
OK, if thats your preference.  Not sure that its really faster unless most of the elements are repeats and there are many of them.

Reply:
Which is often the case for anything other than a bar plot.